### PR TITLE
WT-12005 Fix memory leak in __wt_background_compact_signal

### PR DESCRIPTION
**Pull request by Sean04**

### CODE REVIEW

File: test/cppsuite/tests/background_compact.cpp [hunk 1]
```diff
@@ -300,12 +300,9 @@ class background_compact : public test {
         logger::log_msg(
           LOG_INFO, type_string(tw->type) + " thread {" + std::to_string(tw->id) + "} commencing.");
 
-        bool enabled = false;
-
         while (tw->running()) {
-            enabled = !enabled;
```
Comment (etienneptl) [inline test/cppsuite/tests/background_compact.cpp:304]: Can you add a comment here saying we are doing this on purpose?

File: test/format/compact.c [hunk 2]
```diff
@@ -63,26 +63,21 @@ background_compact(void *arg)
         if (g.workers_finished)
             break;
 
-        if (g.background_compaction_running)
-            testutil_snprintf(config_buf, sizeof(config_buf), "%s", "background=false");
-        else
+        if (mmrand(&g.extra_rnd, 0, 1))
```
Comment (etienneptl) [inline test/format/compact.c:66]: Same here please.

File: test/format/compact.c [hunk 3]
```diff
@@ -63,26 +63,21 @@ background_compact(void *arg)
...
 
     /* Always disable the background compaction server. */
-    if (g.background_compaction_running)
-        ret = session->compact(session, NULL, "background=false");
-    if (ret == 0)
-        g.background_compaction_running = false;
-    else
+    ret = session->compact(session, NULL, "background=false");
+    if (ret != 0)
         testutil_assertfmt(ret == EBUSY, "WT_SESSION.compact failed: %d", ret);
```
Comment (etienneptl) [inline test/format/compact.c:81]: Is that possible to have `EBUSY` here?

Comment (Sean04) [reply test/format/compact.c:81]: Yes, if we haven't process the previous signal, it's possible to return EBUSY.

File: test/cppsuite/tests/background_compact.cpp (hunk 1 above)
Comment (Sean04) [reply test/cppsuite/tests/background_compact.cpp]: I've added
```
    /*
    * The API supports enabling or disabling the background compact server multiple times
    * in a row. Randomly pick whether we are enabling or disabling to cover all state
    * changes.
    */
```

**Review by etienneptl — APPROVED** [review]
LGTM! Good job!

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Comment by Sean04** [comment]
evergreen merge

**Pull request merged**